ui/test/nuxeo-document-comment.test.js

```diff
@@ -0,0 +1,478 @@
+/**
+@license
+(C) Copyright Nuxeo Corp. (http://nuxeo.com/)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import '../nuxeo-document-comments/nuxeo-document-comment.js';
+import {
+  fixture,
+  flush,
+  html,
+  isElementVisible,
+  timePasses,
+  waitForAttrMutation,
+  waitForEvent,
+} from '@nuxeo/nuxeo-elements/test/test-helpers';
+import { pressAndReleaseKeyOn, tap } from '@polymer/iron-test-helpers/mock-interactions';
+import fakeServer from '@nuxeo/nuxeo-elements/test/nuxeo-mock-client';
+
+function getCommentContent(element) {
+  return element.shadowRoot.querySelector('#content');
+}
+
+/* eslint-disable no-unused-expressions */
+suite('<nuxeo-document-comment>', () => {
+  let element;
+  let server;
+
+  setup(async () => {
+    server = fakeServer.create({
+      properties: {
+        username: 'John Doe',
+      },
+    });
+
+    const comment = {
+      parentId: 'parent-id',
+      id: 'comment-id',
+      numberOfReplies: 0,
+      author: 'John Doe',
+      creationDate: '2019-12-09',
+      text: 'This is my testing comment',
+    };
+
+    element = await fixture(
+      html`
+        <nuxeo-document-comment .comment="${comment}"></nuxeo-document-comment>
+      `,
+    );
+  });
+
+  teardown(() => {
+    server.restore();
+  });
+
+  suite('Visibility', () => {
+    test('Should not render any content if no comment data is provided', async () => {
+      element.comment = null;
+      await flush();
+
+      expect(isElementVisible(getCommentContent(element))).to.be.false;
+    });
+
+    test('Should render content if comment data is provided', () => {
+      expect(isElementVisible(getCommentContent(element))).to.be.true;
+    });
+
+    suite("Comment's Content", () => {
+      test('Should not display "show all" option when comment has less than 256 characters', () => {
+        const content = getCommentContent(element);
+        expect(isElementVisible(content.querySelector('span.pointer'))).to.be.false;
+      });
+
+      test('Should display "show all" option when comment has more than 256 characters', async () => {
+        element.set(
+          'comment.text',
+          'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. ' +
```

Comment:
DRY: since we reuse this a bunch of times during this test, we could store in in a variable.